src/Microsoft.ML.Data/Training/TrainerEstimatorBase.cs

```diff
@@ -64,6 +64,7 @@ public abstract class TrainerEstimatorBase<TTransformer, TModel> : ITrainerEstim
             WeightColumn = weight;
         }
 
+        /// <summary> Trains and returns a <see cref="ITransformer"/>.</summary>
```

Comment:
>/// <summary> Trains and returns a <see cref="ITransformer"/>.</summary> [](start = 8, length = 72)

Please mention something like `Derived class can overload this function. For example, in <see cref="FastTreeTrainer">, another <see cref="Fit"> which consumes two data sets, training and validation sets, is implemented`.

Reply:
Why? This is a base class, anyone who derives from it can overload everything.

---
In reply to: [256967851](https://github.com/dotnet/machinelearning/pull/2528#discussion_r256967851) [](ancestors = 256967851)

Reply:
This is for design documentation. Closely related things should be documented explicitly.

---
In reply to: [256974019](https://github.com/dotnet/machinelearning/pull/2528#discussion_r256974019) [](ancestors = 256974019,256967851)

Reply:
It's weird to have references to derived classes in base class especially it it's not functionality of base class. References from top to bottom make sense since you rely on base class and you know what is underneath you. Over way around, I don't see how it make sense. 

---
In reply to: [256976207](https://github.com/dotnet/machinelearning/pull/2528#discussion_r256976207) [](ancestors = 256976207,256974019,256967851)